design/src/main/java/com/urlaunched/android/design/ui/progressbarcontainer/ProgressBarContainer.kt

```diff
@@ -45,6 +48,59 @@ fun ProgressBarContainer(
     }
 }
 
+@Composable
+fun ProgressBarContainer(
+    modifier: Modifier = Modifier,
+    isLoading: Boolean,
+    indicatorColor: Color = ProgressIndicatorDefaults.circularColor,
+    indicatorTrackColor: Color = ProgressIndicatorDefaults.circularTrackColor,
+    progress: Float = 0F,
+    progressSettings: ProgressSettings,
+    content: @Composable () -> Unit
+) {
+    Box(modifier = modifier.fillMaxSize()) {
+        content()
+
+        if (isLoading) {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(Color.Black.copy(alpha = 0.75f))
+                    .clickable(enabled = false, onClick = {}),
+                contentAlignment = Alignment.Center
+            ) {
+                if (progress > 1) {
+                    CircularProgressIndicator(
+                        modifier = Modifier.size(50.dp),
+                        color = indicatorColor,
+                        trackColor = indicatorTrackColor
+                    )
+                } else {
+                    CircularProgressIndicator(
+                        modifier = Modifier.size(50.dp),
+                        progress = { progress },
+                        color = indicatorColor,
+                        trackColor = indicatorTrackColor
+                    )
+                    Text(
+                        text = "${(progress * 100).toInt()}%",
+                        style = progressSettings.mainTextStyle,
+                        color = progressSettings.mainColor
+                    )
+                    if (progressSettings.description?.isNotEmpty() == true) {
+                        Text(
+                            text = progressSettings.description,
+                            modifier = Modifier.padding(top = 90.dp),
+                            style = progressSettings.descriptionTextStyle,
+                            color = progressSettings.descriptionColor
+                        )
+                    }
+                }
```

Comment:
If the elements supposed to be laid out one after another, maybe using Column will be better approach then increasing padding here? As this will look incorrect if user changes their device's font scale.

Reply:
The problem is that I would need to use Column and then put inside a Box for `CircularProgressIndicator` and the previous Text. Because the percentage should be inside the spinner. If it's fine for you, I'll do it this way. I added the padding to avoid using an additional layout.

Reply:
Have this ui been approved by anybody? It feels weird to me to put percentage inside the spinner

Reply:
Well, if you [look online](https://www.google.com/search?q=circular+progress+percentage+android&sca_esv=e592e3a30352db65&sca_upv=1&rlz=1C5CHFA_enIT1017IT1017&udm=2&biw=2560&bih=1328&ei=uGLgZoChD4-W9u8P5IbdqAI&ved=0ahUKEwiA3d_S1biIAxUPi_0HHWRDFyUQ4dUDCBA&uact=5&oq=circular+progress+percentage+android&gs_lp=Egxnd3Mtd2l6LXNlcnAiJGNpcmN1bGFyIHByb2dyZXNzIHBlcmNlbnRhZ2UgYW5kcm9pZEjwC1C3BViYC3ABeACQAQCYAWSgAe8EqgEDOC4xuAEDyAEA-AEBmAICoAJnwgIGEAAYCBgewgIIEAAYBxgIGB6YAwCIBgGSBwMxLjGgB5UE&sclient=gws-wiz-serp) it's not that unusual. I've tried to find something on the material guidelines but there is no clear indication on how to do it.

However, Vincenzo approved it and said it doesn't require further refinement by designers.